7.tcga-classify/classify.py

```diff
@@ -0,0 +1,233 @@
+filter_count = 15
+folds = 5
+algorithms = ["pca", "ica", "nmf", "dae", "vae"]
+signals = ["signal", "shuffled"]
+alphas = [0.1, 0.13, 0.15, 0.2, 0.25, 0.3]
+l1_ratios = [0.15, 0.16, 0.2, 0.25, 0.3, 0.4]
+
+# Load data to build y matrices
+base_url = "https://github.com/greenelab/pancancer/raw"
+commit = "2a0683b68017fb226f4053e63415e4356191734f"
+
+# Load data
+file = "{}/{}/data/sample_freeze.tsv".format(base_url, commit)
+sample_freeze_df = pd.read_table(file, index_col=0)
+
+file = "{}/{}/data/pancan_mutation_freeze.tsv.gz".format(base_url, commit)
+mutation_df = pd.read_table(file, index_col=0)
+
+file = "{}/{}/data/copy_number_loss_status.tsv.gz".format(base_url, commit)
+copy_loss_df = pd.read_table(file, index_col=0)
+
+file = "{}/{}/data/copy_number_gain_status.tsv.gz".format(base_url, commit)
+copy_gain_df = pd.read_table(file, index_col=0)
+
+file = "{}/{}/data/mutation_burden_freeze.tsv".format(base_url, commit)
+mut_burden_df = pd.read_table(file, index_col=0)
+
+# What results to track
+full_metrics_list = []
+full_auc_list = []
+full_aupr_list = []
+full_coef_list = []
+all_failed_params = []
+
+# Obtain a list of locations for each feature matrix (X)
+z_matrix_dict = {}
+for signal in signals:
+    z_matrix_dict[signal] = {}
+
+    if signal == "signal":
+        results_dir = "TCGA_results"
+    else:
+        results_dir = "TCGA_shuffled_results"
+
+    matrix_dir = os.path.join(
+        "..", "2.ensemble-z-analysis", "results", results_dir, "ensemble_z_matrices"
+    )
+
+    for comp_dir in os.listdir(matrix_dir):
+        matrix_comp_dir = os.path.join(matrix_dir, comp_dir)
+        z_dim = comp_dir.split("_")[2]
+        z_matrix_dict[signal][z_dim] = {}
+
+        for z_file in glob.glob("{}/*_z_*".format(matrix_comp_dir)):
+            seed = os.path.basename(z_file).split("_")[1]
+
+            if seed not in z_matrix_dict[signal][z_dim].keys():
+                z_matrix_dict[signal][z_dim][seed] = {}
+
+            if "_test_" in z_file:
+                z_matrix_dict[signal][z_dim][seed]["test"] = z_file
+            else:
+                z_matrix_dict[signal][z_dim][seed]["train"] = z_file
+
+for gene in genes:
+    # Process the y matrix for the given gene or pathway
+    if gene == "RAS":
+        subset_gene = ["KRAS", "HRAS", "NRAS"]
+        y_copy_number_df = copy_gain_df.loc[:, subset_gene].max(axis="columns")
+        y_mutation_df = mutation_df.loc[:, subset_gene].max(axis="columns")
+    else:
+        y_copy_number_df = copy_loss_df.loc[:, gene]
+        y_mutation_df = mutation_df.loc[:, gene]
+
+    y_df = process_y_matrix(
+        y_copy=y_copy_number_df,
+        y_mutation=y_mutation_df,
+        gene=gene,
+        sample_freeze=sample_freeze_df,
+        mutation_burden=mut_burden_df,
+        filter_count=filter_count,
+        filter_prop=filter_prop,
+        hyper_filter=5,
+    )
+
+    # Now, perform all the analyses for each X matrix
+    for signal in z_matrix_dict.keys():
```

Comment:
Are you testing to see if for each data type (copy number, mutation that is listed at the top) if their compressed form is able to generate the RAS gene expression activity?

Reply:
I am testing to see if the compressed gene expression features can be used to predict pathway status in tumors.

E.g. compressing TCGA gene expression into 50 principal components can predict Ras pathway activation in tumors with a 75% AUROC. The predictions are based on if the sample has a copy number gain or non silent mutation in KRAS, HRAS, or NRAS